generators/generator-base.js

```diff
@@ -46,8 +46,8 @@ Generator.prototype.addElementToMenu = function (routerName, glyphiconName, enab
             file: fullPath,
             needle: 'jhipster-needle-add-element-to-menu',
             splicable: [
-                '<li ui-sref-active="active" >\n' +
-                '                    <a ui-sref="' + routerName + '" ng-click="vm.collapseNavbar()">\n' +
+                '<li uiSrefActive="active" >\n' +
```

Comment:
Yes, protractor will work, you just have to use `by.id` and `by.css`. You can see https://github.com/mraible/ng2-demo for examples.